### DISCUSSION TITLE
How to deal with large saccades?

### DISCUSSION THREAD

**Discussion by AnaisServais** [unfoldtoolbox/unfold#136] [Q&A] [answered]
Hello,

First of all, thank you for this great toolbox!

I am a postdoctoral researcher and I would like to carry out a co-registration study with EEG and eye-tracking (Eyelink 1000 remote mode) to investigate the relationship between the attentional switch and gaze aversion. I am interested in using your _unfold toolbox_ since saccades are assumed to have different sizes and directions between my experimental conditions. However, I am facing an issue since gaze aversion is eye movement with a particularly large amplitude (so much so that the gaze exits the screen causing eye-tracking signal losses).

I, therefore, have **two questions**:
1) Does the toolbox perform well for large amplitude saccades? Is there a maximal saccade size to respect in order to use the toolbox?
2) How does the toolbox deal with eye-tracking signal losses (knowing that I also use EOG electrodes)? Indeed, the gaze aversion I want to study tends to generate signal loss (because the gaze exits the eye-tracker tracking field). 

Thank you in advance for your help.
Best regards,
Anaïs

**Comment by olafdimigen** [comment, marked as answer]
Hi Anais,

question (1): the toolbox performs generally well with small or large saccades, there is no upper limit. Larger saccades of course generate larger ocular artifacts (which can be removed with ICA-based procedures) and also larger brain responses (e.g., see Figure in Dimigen & Ehinger, 2021, _J Vis_) and there is a nonlinear relationship between saccade size and brain response amplitude, which is why it can be useful to add saccade size as a nonlinear spline predictor: `saccERP ~ 1 + spl(saccSize,4)`.

One thing that is important is that the unfoldtoolbox only needs some distinct events (usually onsets) telling it when your saccade happens and how large it was, in contrast, it does not need your continuous eye-tracking signal for standard models. So your problem is mostly how to get good events out of ET data with missing samples (once the eye goes outside of the measurement range of your eye-tracker), not so much about unfold modeling per se.
 
I also assume that if you switch on online saccade detection in Eyelink, it will still give you the saccade-onset events for out-of-screen saccades, but I personally have not tried that (note: an out-of-screen saccade may or may not be still within the EL measurement range, the system should extrapolate beyond the range for which it was calibrated, so you may still get measurements even for gaze positions slightly beyond the screen). Alternatively, you can also just get the onset events for your out-of-range saccades from your EOG signals (you could also use an amplitude estimate from there, e.g. in µV).

While your unfold model will be better if you take saccade size into account, it is much more important to model the saccade event at all even if you can only use its intercept term `(y ~ 1)` than to leave the event out of the model. If you have in addition lots of properly tracked smaller within-monitor-range saccades and then a separate, sufficiently large population of out-of-range saccades, it can also be a good solution to define two distinct predictors for both saccade types, like this:

```
stimulusERP ~ 1 + cat(myFactorOfInterest)  % for your stimulus onset events, e.g. picture onset
saccade_normal_ERP ~ 1 + saccAmplitude  % for your "normal" (micro)saccade events, for which you have saccade amplitudes
% optionally use a spline predictor instead above, i.e. 1 + spl(saccAmplitude,4)
saccade_outside_ERP ~ 1 % for your out-of-range saccades, intercept only
```

TL;DR; answer to your questions:
1. yes, performs well; there is no max. saccade amplitude
2. the toolbox does not need your raw ET data for standard models, just events, so ET loss is not an issue per se, only for your event detection algorithm

Hope this helps,
Olaf

**Comment by AnaisServais** [comment]
Hi Olaf,

Thanks a lot for your quick and very detailed response.
So, if only saccades events (onset and duration) are required, could these events be extracted from the EOG signal instead of the eye-tracking one (at least for the moments where the eye-tracking signal is lost)?

Best,
Anaïs

**Comment by olafdimigen** [comment]
Yes, you can also use events extracted from your EOG signal. If the properties of these events are different from those extracted from the eye-tracking data, you can consider using slightly different predictors for that event category, as I have laid out above with my (sub)formulas, where EOG-derived saccades do not have a saccade amplitude. Of course you can also try to estimate the saccade amplitude of your EOG-based saccade events (note that saccade amplitude is not the same as saccade duration, but they are of course highly correlated).
